decidim-core/app/controllers/concerns/decidim/devise_controllers.rb

```diff
@@ -18,6 +18,7 @@ module DeviseControllers
       include ImpersonateUsers
       include NeedsPermission
       include Decidim::SafeRedirect
+      include NeedsSnippets
```

Comment:
Yes, Devise controllers also need it because they use the same layout as other views in Decidim.

And in the layout's `_head.html.erb` the `snippets` object is needed:
https://github.com/mainio/decidim/blob/a0da97a469f3beef3db91d22ba58ba4ce900164c/decidim-core/app/views/layouts/decidim/_head.html.erb#L30

In case the concern was not included in the `DeviseControllers`, all Devise views would throw an exception (undefined local variable or method) which would block the users e.g. from signing in or signing up to the service.

Reply:
mmm.. there was the option to use `<%= snippets.display(:head) if defined?(snippets) %> to get rid of this problem no? isn't it better to have it optional?

Reply:
Yes, that's definitely another option. But I'd prefer this to be always available because it has potential for other use cases, e.g. script injections from engines. Sort of like you can expect the `current_organization` to be always available.

One use case could be e.g. implementing [a custom cookie banner](https://github.com/OpenSourcePolitics/decidim-module_cookies).

I know there are alternative ways of implementing that example as well (as shown in that module) but my point is that if this kind of "middle layer" is added because the developer of `trailblazer/cells` doesn't like the default implementation in Rails, this should be as important part as the `cell` method itself.

I'd prefer obviously to use `content_for` for this but it's not possible with cells.